test/cmdlineTests/recovery_ast_constructor/output

```diff
@@ -58,6 +58,7 @@ JSON AST:
         {
           "attributes":
           {
+            "freeFunction": false,
             "implemented": true,
             "isConstructor": true,
             "kind": "constructor",
```

Comment:
Why do we have `isConstructor` if we have kind? And why introduce a new attribute `freeFunction` instead of extending `kind`?

Reply:
I didn't see "being free" and the "kind of function" to be the same thing, but you are right that they currently all exclude each other.